cmd/soroban-rpc/internal/methods/get_ledgers.go

```diff
@@ -0,0 +1,252 @@
+			return fmt.Errorf("limit must not exceed %d", maxLimit)
+		}
+	case req.StartLedger != 0 && !isStartLedgerWithinBounds(req.StartLedger, ledgerRange):
+		return fmt.Errorf(
+			"start ledger must be between the oldest ledger: %d and the latest ledger: %d for this rpc instance",
+			ledgerRange.FirstLedger.Sequence,
+			ledgerRange.LastLedger.Sequence,
+		)
+	}
+
+	return IsValidFormat(req.Format)
+}
+
+// LedgerInfo represents a single ledger in the response.
+type LedgerInfo struct {
+	Hash            string `json:"hash"`
+	Sequence        uint32 `json:"sequence"`
+	LedgerCloseTime int64  `json:"ledgerCloseTime,string"`
+
+	LedgerHeader     string          `json:"headerXdr"`
+	LedgerHeaderJSON json.RawMessage `json:"headerJson,omitempty"`
+
+	LedgerMetadata     string          `json:"metadataXdr"`
+	LedgerMetadataJSON json.RawMessage `json:"metadataJson,omitempty"`
+}
+
+// GetLedgersResponse encapsulates the response structure for getLedgers queries.
+type GetLedgersResponse struct {
+	Ledgers               []LedgerInfo `json:"ledgers"`
+	LatestLedger          uint32       `json:"latestLedger"`
+	LatestLedgerCloseTime int64        `json:"latestLedgerCloseTime"`
+	OldestLedger          uint32       `json:"oldestLedger"`
+	OldestLedgerCloseTime int64        `json:"oldestLedgerCloseTime"`
+	Cursor                string       `json:"cursor"`
+}
+
+type ledgersHandler struct {
+	ledgerReader db.LedgerReader
+	maxLimit     uint
+	defaultLimit uint
+}
+
+// NewGetLedgersHandler returns a jrpc2.Handler for the getLedgers method.
+func NewGetLedgersHandler(ledgerReader db.LedgerReader, maxLimit, defaultLimit uint) jrpc2.Handler {
+	return NewHandler((&ledgersHandler{
+		ledgerReader: ledgerReader,
+		maxLimit:     maxLimit,
+		defaultLimit: defaultLimit,
+	}).getLedgers)
+}
+
+// getLedgers fetch ledgers and relevant metadata from DB.
+func (h ledgersHandler) getLedgers(ctx context.Context, request GetLedgersRequest) (GetLedgersResponse, error) {
+	readTx, err := h.ledgerReader.NewTx(ctx)
+	if err != nil {
+		return GetLedgersResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InternalError,
+			Message: err.Error(),
+		}
+	}
+	defer func() {
+		_ = readTx.Done()
+	}()
+
+	ledgerRange, err := readTx.GetLedgerRange(ctx)
+	if err != nil {
+		return GetLedgersResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InternalError,
+			Message: err.Error(),
+		}
+	}
+
+	if err := request.validate(h.maxLimit, ledgerRange); err != nil {
+		return GetLedgersResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidRequest,
+			Message: err.Error(),
+		}
+	}
+
+	start, limit, err := h.initializePagination(request, ledgerRange)
+	if err != nil {
+		return GetLedgersResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	ledgers, err := h.fetchLedgers(ctx, start, limit, request.Format, readTx)
+	if err != nil {
+		return GetLedgersResponse{}, err
+	}
+	cursor := strconv.Itoa(int(ledgers[len(ledgers)-1].Sequence))
```

Comment:
is it possible for ledgers to be empty ? Consider the case where you query the latest ledgers and use the cursor to query the next page. In that scenario, you have exhausted all the ledgers